control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -719,6 +718,38 @@ func (r *reconciler) reconcileIngressController(ctx context.Context, hcp *hyperv
 			errs = append(errs, fmt.Errorf("failed to reconcile default ingress controller cert: %w", err))
 		}
 	}
+
+	// Default Ingress is passed through as a subdomain of the infra/mgmt cluster
+	// for KubeVirt when the base domain passthrough feature is in use.
+	baseDomainPassthrough := hcp.Annotations[hyperv1.KubeVirtBaseDomainPassthroughAnnotation]
+	if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform && baseDomainPassthrough == "true" {
+		defaultIngressNodePortService := manifests.IngressDefaultIngressNodePortService()
+		err := r.client.Get(ctx, client.ObjectKeyFromObject(defaultIngressNodePortService), defaultIngressNodePortService)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to retrieve guest cluster ingress NodePort: %w", err))
+		}
+
+		hcpNamespace := hcp.Namespace
+
+		// Manifests for infra/mgmt cluster passthrough service
+		cpService := manifests.IngressDefaultIngressPassthroughService(hcpNamespace)
+
+		// Manifests for infra/mgmt cluster passthrough routes
+		cpPassthroughRoute := manifests.IngressDefaultIngressPassthroughRoute(hcpNamespace)
+
+		if _, err := r.CreateOrUpdate(ctx, r.cpClient, cpService, func() error {
+			return ingress.ReconcileDefaultIngressPassthroughService(cpService, defaultIngressNodePortService, hcp)
+		}); err != nil {
+			errs = append(errs, fmt.Errorf("failed to reconcile kubevirt ingress passthrough service: %w", err))
+		}
+
+		if _, err := r.CreateOrUpdate(ctx, r.cpClient, cpPassthroughRoute, func() error {
+			return ingress.ReconcileDefaultIngressPassthroughRoute(cpPassthroughRoute, cpService, hcp)
+		}); err != nil {
+			errs = append(errs, fmt.Errorf("failed to reconcile kubevirt ingress passthrough route: %w", err))
+		}
+	}
+
```

Comment:
I'm creating a route and service in the hosted control plane namespace while in the HCCO (which typically only works on the guest client, not the mgmt client).

This is being done because I have to get the NodePort's port used by the default ingress services in the guest cluster in order to create the service in the mgmt cluster. basically, the mgmt cluster service has to point the backend "somewhere", and that somewhere is the nodeport's port of the routers in the guest cluster.

Reply:
done